android-app/app/src/main/java/de/dhbw/ka/se/fibo/ui/dashboard/DashboardFragment.java

```diff
@@ -87,38 +103,82 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         }
     }
 
-    private void initializePieChart() {
+    private void populateCategories() {
         Context context = requireContext();
-        SortedSet<Cashflow> cashflows = ApplicationState.getInstance(context).getCashflows();
 
-        Stream<Cashflow> cashflowStream = cashflows.stream().filter(x -> CashflowType.EXPENSE == x.getType());
+        orderedCategoryList = ApplicationState.getInstance(context)
+                .getCashflows()
+                .stream()
+                .filter(x -> CashflowType.EXPENSE == x.getType())
+                .map(Cashflow::getCategory)
+                .collect(Collectors.toSet()) // make sure we have only one occurrence of each category
+                .stream() // then sort by localized name of the category
+                .sorted(Comparator.comparing(o -> context.getText(o.getName()).toString()))
+                .collect(Collectors.toList());
+    }
 
-        if (null != startDate) {
-            cashflowStream = cashflowStream.filter(x -> startDate.minusDays(1).isBefore(ChronoLocalDate.from(x.getTimestamp())));
-        }
+    private void initializeFilter() {
+        Button filterButton = binding.openFilterOptions;
+        filterButton.setOnClickListener(v -> {
+            Log.i("FiBo", "clicked filter button");
 
-        if (null != endDate) {
-            cashflowStream = cashflowStream.filter(x -> endDate.plusDays(1).isAfter(ChronoLocalDate.from(x.getTimestamp())));
-        }
+            Context context = requireContext();
 
-        Map<Category, BigDecimal> expensesPerCategory = new HashMap<>();
+            boolean[] checkedItems = new boolean[orderedCategoryList.size()];
+            CharSequence[] names = new CharSequence[orderedCategoryList.size()];
 
-        for (Cashflow cashflow : cashflowStream.collect(Collectors.toList())) {
-            Category category = cashflow.getCategory();
+            for (int i = 0; i < checkedItems.length; i++) {
+                Category category = orderedCategoryList.get(i);
 
-            BigDecimal newValue = expensesPerCategory
-                    .computeIfAbsent(category, x -> BigDecimal.ZERO)
-                    .add(cashflow.getOverallValue());
+                checkedItems[i] = !hiddenCategories.contains(category);
+                names[i] = context.getText(category.getName());
```

Comment:
isn't this a good usecase for a Map of checked Items instead of using 2 arrays?

Reply:
no, because the Android API excepts two arrays and you'd need to keep track of all the categories and all the ones that are hidden either way